progressions.rb

```diff
@@ -2,5 +2,27 @@ def geometric(a, q, n)
   a * q ** (n-1)
 end
 
+class Fib
+  def initialize
+    @cache = {
+      1 => 1,
+      2 => 1
+    }
+  end
+
+  def fib(n)
+    @cache[n] ||= fib(n-1) + fib(n-2)
+  end
+end
```

Comment:
Świetnie! Fajny pomysł. Ale OO w tym miejscu, to chyba trochę przerost formy nad treścią. Nie sądzisz, że jednak prostszym rozwiązaniem będzie napisanie tego iteracyjnie?